Documentation/conf.py

```diff
@@ -37,6 +37,7 @@
 
 # -- Project information -----------------------------------------------------
 
+master_doc = 'index'
```

Comment:
why is this being done here?

Reply:
https://stackoverflow.com/questions/56336234/build-fail-sphinx-error-contents-rst-not-found

Reply:
While this is the readthedocs theme it is not the same as hosting on readthedocs.org.
I have not encountered any issue and CI build documentation correctly. Can you copy the error you are receiving?
Also, please verify you are using the expected versions of all sphinx packages. The safest why is doing it like it is described here: https://nuttx.apache.org/docs/latest/contributing/documentation.html

Reply:
Sphinx error:
master file /home/ligd/platform/mirtos3/nuttx/Documentation/contents.rst not found
Makefile:42: recipe for target 'html' failed
make: *** [html] Error 2

Maybe should use the install cmds as your recommended, but mine setup the sphinx as follows:
sudo  pip install m2r2
sudo apt-get install python-pip
pip install -U sphinx
sudo pip install sphinx_rtd_theme

This is caused by the conflicts of the default sphinx version in readthedocs and the local environment.
See:
https://github.com/readthedocs/readthedocs.org/issues/2569

I think we should cover this situation.

Reply:
But again, we're not using that service. We're simply using their theme.
The point is that we have a working set of package versions for all these dependencies so that we agree on a working version. Please follow the approach in the documentation and see if you still get the error.

Reply:
sphinx-build 1.8.5 will meet compile error:
master file /home/ligd/platform/mirtos3/nuttx/Documentation/contents.rst not found

after update to sphinx-build 3.3.0+:
then compile pass

remove the modify as you like.

Reply:
Great, thanks